kalite/topic_tools/content_models.py

```diff
@@ -416,19 +427,15 @@ def recurse_availability_up_tree(node, available):
                     recurse_availability_up_tree(parent, available)
 
 
-            for path, update in updates_dict.iteritems():
+            for path, update in content_models:
                 if update:
                     # We have duplicates in the topic tree, make sure the stamping happens to all of them.
-                    items = Item.select().where(Item.path == path)
+                    
+                    items = Item.select().where((Item.path == path) & (Item.kind != "Topic"))
+                    Item.update("available", True).where((Item.path == path) & (Item.kind != "Topic"))
```

Comment:
You are updating things irregardless of the contents of ```update``` here. It is possible that something should be flagged as unavailable.

Also, there is other data stored in update, aside from availability.